objects/aidBoard/script.lua

```diff
@@ -849,45 +849,17 @@ function toggleElements()
 end
 
 function scanElements()
-    local elemCardTable = {}
-    for _,guid in pairs(Global.getVar("elementScanZones")) do
-        local zone = getObjectFromGUID(guid)
-        --Go through all items found in the zone
-        for _, entry in ipairs(zone.getObjects()) do
-            --Ignore non-cards
-            if entry.type == "Card" then
-                --Ignore if no elements entry
-                if entry.getVar("elements") ~= nil then
-                    if not entry.is_face_down and entry.getPosition().z > zone.getPosition().z then
-                        table.insert(elemCardTable, entry)
-                    end
-                end
-            end
+    local elements = {0,0,0,0,0,0,0,0}
+    for _, selected in pairs(Global.getVar("selectedColors")) do
+        for i, _ in ipairs(elements) do
+            elements[i] = elements[i] + selected.elements[i].getButtons()[1].label
         end
     end
-    local combinedElements = elemCombine(elemCardTable)
-    local elementsTable = {"Sun","Moon","Fire","Air","Water","Earth","Plant","Animal"}
-    for i,total in ipairs(combinedElements) do
+    local elementNames = {"Sun","Moon","Fire","Air","Water","Earth","Plant","Animal"}
+    for i, total in ipairs(elements) do
         getObjectFromGUID(elementGuids[i]).editButton({
             index = 0,
-            label = total + #getObjectsWithTag(elementsTable[i]),
+            label = total + #getObjectsWithTag(elementNames[i]),
```

Comment:
this looks to be double counting element markers that are in player scripting zones

Reply:
Yeah, fixed. I guess I must have tunneled on thing that was getting the button to update, and not being part of the total, and not actually read the code. :shrug:

Reply:
now this doesn't appear to be counting elements placed on the island (elemental invocation)

Reply:
Gah. I'll look at that later today. I'm inclined to adjust it so you don't need to double the tokens on the board to get the tracking for them right, though.

Reply:
i'm not against a solution to make it so you don't need to double up, but i also don't want to introduce more physics casting for every single element token to determine if it's on the island, player table, or somewhere else

Reply:
maybe when the element token collides with the island board it adds a special tag that makes that count twice?

Reply:
My initial thought was to just upcast from the sea tile, and grab all the tokens, and we can just double their elements. (Or, I don't recall if there is a scripting zone covering the sea tile, if there is, we can just check in there, instead of casting).

Reply:
this is not a scripting zone covering the sea tile because there's no current need for it, and then if you try to use gizmo tool to move something like island board it's pretty unusable now

Reply:
Yeah, a single box cast with distance 0 is able to handle it.